flake.nix

```diff
@@ -32,6 +32,7 @@
           "rust-src"
           "rustc"
           "rustfmt"
+          "rust-analyzer"
```

Comment:
Extensions in VSCode may not even need which might be why it isn't here

Reply:
resolving because i cant merge otherwise